build/build/src/engine/context.rs

```diff
@@ -216,6 +216,36 @@ impl RunContext {
         Ok(())
     }
 
+    /// During the native-image build, the engine generates arg files. This function uploads them as
+    /// artifacts on the CI, so we can inspect them later.
+    /// Note that if something goes wrong, the native image arg files may not be present.
+    async fn upload_native_image_arg_files(&self) -> Result {
+        debug!("Uploading Native Image Arg Files");
+        let engine_runner_ni_argfile =
+            self.repo_root.engine.runner.target.native_image_args_txt.path.clone();
+        let launcher_ni_argfile =
+            self.repo_root.engine.launcher.target.native_image_args_txt.path.clone();
+        let project_manager_ni_argfile =
+            self.repo_root.lib.scala.project_manager.target.native_image_args_txt.path.clone();
+        let native_image_arg_files: HashMap<&PathBuf, &str> = HashMap::from_iter([
```

Comment:
HashMap is probably not needed here, we should be able to just assign list literal with pairs directly.

Reply:
HashMap construction removed in 50692c515ca4f416fec1b0bc1fcb033b0f836056